genie-agent/src/main/java/com/netflix/genie/agent/AgentMetadata.java

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent;
+
+import org.hibernate.validator.constraints.NotBlank;
+
+/**
+ * Container for Genie agent metadata and runtime information.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+public interface AgentMetadata {
```

Comment:
Why an interface and not just a class?

Reply:
So the implementation is package-local. I though you said you wanted to to more of this moving forward, rather than just making classes public willy-nilly because they're used in a different package.